src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -59,6 +60,7 @@ public class ConnectionService
         private readonly ConcurrentDictionary<CancelTokenKey, CancellationTokenSource> cancelTupleToCancellationTokenSourceMap =
                     new ConcurrentDictionary<CancelTokenKey, CancellationTokenSource>();
 
+        public Dictionary<string, TokenState> IsConnected = new Dictionary<string, TokenState>();

```

Comment:
`IsConnected` isn't a very clear name, and I'm not sure why you'd need separate states anyways. Just a set of URIs currently being refreshed should be fine, you add it when a refresh is requested and then remove it when the token refreshed notification is received.